util/db/db.go

```diff
@@ -29,7 +32,12 @@ type ArgoDB interface {
 	// CreateRepository creates a repository
 	CreateRepository(ctx context.Context, r *appv1.Repository) (*appv1.Repository, error)
 	// GetRepository returns a repository by URL
-	GetRepository(ctx context.Context, name string) (*appv1.Repository, error)
+	GetRepository(ctx context.Context, repoURL string) (*appv1.Repository, error)
+	// GetRepositoryCredential will return a credential for use with the repository, or an error if not found.
+	// The returned credential generally won't have the same URL.
+	GetRepositoryCredential(ctx context.Context, repoURL string) (*appv1.Repository, error)
+	// HydrateRepositoryCredentials
+	HydrateRepositoryCredentials(ctx context.Context, repo *appv1.Repository) error
```

Comment:
I've noticed that `HydrateRepositoryCredentials` is always executed right after `GetRepository`. e.g.

```
repo, err := s.db.GetRepository(ctx, repoURL)
if err != nil {
    // If we couldn't retrieve from the repo service, assume public repositories
    repo = &appv1.Repository{Repo: repoURL}
}

_ = s.db.HydrateRepositoryCredentials(ctx, repo)
```


I think it makes sense to encapsulate HydrateRepositoryCredentials logic in GetRepository method. In this case you can remove make `GetRepositoryCredential` and `HydrateRepositoryCredentials` methods private and reduce amount of changes.

Reply:
I did consider this, but I'm in two minds about it. It would change the semantics of the method and so that `GetRepository` and `GetRepository` would not be symmetric. You could accidentally copy the credentials into the repository. Which would be annoying

Reply:
Ok. So I do need to be able to call `GetRepository` without populating the credentials. So I introduced `GetHydratedRepository`, which does that. An alternative would be to have add  `hydrated bool` parameter to `GetRepository`.

Thoughts?